firebase.json

```diff
@@ -5,8 +5,20 @@
     "trailingSlash": false,
     "headers": [{
         "source": "**/*.@(txt|json)",
-        "headers": [ { "key": "Access-Control-Allow-Origin", "value": "*" } ]
-    }],
+        "headers": [
+          { "key": "Access-Control-Allow-Origin", "value": "*" }
+        ]
+    },
+      {
+        "source": "**",
+        "headers": [
+          { "key": "Content-Security-Policy", "value":  "default-src https:; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://www.googletagmanager.com https://www.google-analytics.com https://ssl.google-analytics.com https://gstatic.com https://*.gstatic.com https://dartpad.dev https://*.dartpad.dev https://youtube.com https://fonts.googleapis.com https://doubleclick.net https://*.doubleclick.net https://google.com https://*.google.com https://storage.googleapis.com; object-src 'none'; base-uri 'none'; style-src https: 'unsafe-inline'"},
```

Comment:
Why do we need: `'unsafe-inline' 'unsafe-eval'`

I'm guess there is an actual reason :D

Perhaps we file a follow-up to fix this.

Reply:
The [search page](https://dart.dev/search) and the Google Tag Manager script on every page relied on `unsafe-inline`. Adding the hashes, I was able to remove `unsafe-inline`, and as far as I can there are no new problems popping up from that change.

`unsafe-eval` seems trickier as it is currently used by Google CSP for the search page as well as on the [Dart SDK Archive](https://dart.dev/tools/sdk/archive) page.

Reply:
@parlough another option, instead of adding hashes is to embed the Google Tag Manager script from a `.js` file.

See:
https://github.com/dart-lang/pub-dev/blob/master/static/js/gtm.js

That's how we did on pub.dev